src/Chip/Chip.js

```diff
@@ -0,0 +1,168 @@
+// @flow weak
+
+import React, { PropTypes } from 'react';
+import { createStyleSheet } from 'jss-theme-reactor';
+import classNames from 'classnames';
+import DeleteIcon from '../svg-icons/navigation/cancel';
+
+export const styleSheet = createStyleSheet('Chip', (theme) => {
+  const { palette, shadows, transitions } = theme;
+  return {
+    root: {
+      fontSize: 13,
+      display: 'flex',
+      alignItems: 'center',
+      justifyContent: 'center',
+      height: 32,
+      outline: 'none',
+      border: 0,
+      color: palette.getContrastText(palette.grey[300]),
+      backgroundColor: palette.grey[300],
+      boxShadow: shadows[0],
+      borderRadius: 16,
+      whiteSpace: 'nowrap',
+      width: 'fit-content',
+      transition: transitions.create(),
+    },
+    clickable: {
+      '&:active': {
+        boxShadow: shadows[1],
+      },
+      '&:hover, &:active': {
+        backgroundColor: palette.grey.A100,
+      },
+      cursor: 'pointer',
+    },
+    avatar: {
+      marginRight: -4,
+      width: 32, // Added for jss avatar
+      height: 32, // Added for jss avatar
+      fontSize: 16,
+    },
+    avatarIcon: {
+      width: 19.2, // Added for jss avatar
+      height: 19.2, // Added for jss avatar
+    },
+    label: {
+      color: palette.text,
+      // display: 'inline-flex', // For broken avatar
+      // alignItems: 'center', // For broken avatar
+      paddingLeft: 12,
+      paddingRight: 12,
+      userSelect: 'none',
+      whiteSpace: 'nowrap',
+    },
+    deleteIcon: {
+      fill: palette.grey[500],
+      '&:hover': {
+        fill: palette.grey[600],
+      },
+      cursor: 'pointer',
+      margin: '0 4px 0 -8px',
+    },
+  };
+});
+
+/**
+ * Chips represent complex entities in small blocks, such as a contact.
+ *
+ * ```jsx
+ * <Chip>
+ *   <Avatar>
+ *   Label text
+ * </Chip>
+ * ```
+ */
+export default function Chip(props, context) {
+  const {
+    children: childrenProp,
+    className: classNameProp,
+    deleteIconClassName: deleteIconClassNameProp,
+    labelClassName: labelClassNameProp,
+    onRequestDelete,
+    onClick,
+    ...other
+  } = props;
+
+  let children = childrenProp;
+
+  const classes = context.styleManager.render(styleSheet);
+  const className = classNames(classes.root, { [classes.clickable]: onClick }, classNameProp);
+  const avatarClassName = classNames(classes.avatar);
```

Comment:
classNames is an helper. That has nothing to do with jss. https://github.com/JedWatson/classnames.